Engine/ac/timer.cpp

```diff
@@ -82,19 +87,25 @@ void WaitForNextFrame()
 
     auto frame_time_remaining = next_frame_timestamp - now;
     if (frame_time_remaining > std::chrono::milliseconds::zero()) {
+#if AGS_PLATFORM_OS_EMSCRIPTEN
```

Comment:
Question: should not this logically check for AGS_DISABLE_THREADS?
if not, what is the reason that Emscripten can work with SDL_Delay but not std:: thread sleep?

Reply:
Theoretically both should work, [Asyncify](https://youtu.be/qQOP6jqZqf8) should work with either of them - this is the spot Asyncify  should unroll the stack, allow the browser to do it's work and come back. But for some reason the result with std::thread currently is the browser getting unresponsive - which is weird since it's written by Emscripten people. [SDL_Delay](https://github.com/libsdl-org/SDL/blob/main/src/timer/SDL_timer.c) appears to use some JS wait function when running with Emscripten, and it's working.

Reply:
Also, I forgot, you can enable threads which will run the Sound thread through a JS Worker, but the "main" thread runs on the browser and still requires the browser to have some time to draw and process inputs. So the SDL_Delay is just because it's Emscripten, but the sounds parts are the difference between with thread and without threads.